flink-java/src/main/java/org/apache/flink/api/java/typeutils/TypeExtractor.java

```diff
@@ -391,6 +392,10 @@ protected TypeExtractor() {
 				curT = typeToClass(curT).getGenericSuperclass();
 			}
 			
+			if(typeToClass(curT).equals(Tuple0.class)) {
```

Comment:
`curT` could be `TypeVariable` or someting else at this point. I think it's better to use `if (curT instanceof Class<?> && curT == Tuple0.class)`